### PR TITLE
(cloudwatch) add "Divide Sum By Period" option

### LINKED ISSUE #3187: CloudWatch plugin can't handle DynamoDB correctly
Current implementation can't draw correct graph for ConsumedReadCapacityUnits and ConsumedWriteCapacityUnits.

According to document, the plugin should divide Sum by 60 to getting correct Average value.

http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/MonitoringDynamoDB.html

> Use the Sum statistic to calculate the consumed throughput. For example, get the Sum value over a span of 1 minute, and divide it by the number of seconds in a minute (60) to calculate the average ConsumedReadCapacityUnits per second (recognizing that this average will not highlight any large but brief spikes in read activity that occurred during that minute). You can compare the calculated value to the provisioned throughput value you provide DynamoDB.

### PR DESCRIPTION
**Pull request by mtanda**
Another approach to fix https://github.com/grafana/grafana/issues/3187 .
There are not so much exceptional cases which we need to calculate in Grafana side.
And the expression is only "Sum / Period", so I simply add checkbox to use the expression.

These are exceptional cases what I found.

DynamoDB: ConsumedReadCapacityUnits, ConsumedWriteCapacityUnits
https://docs.aws.amazon.com/AmazonCloudWatch/latest/DeveloperGuide/dynamo-metricscollected.html

EBS: VolumeReadBytes, VolumeWriteBytes, VolumeReadOps, VolumeWriteOps
https://docs.aws.amazon.com/AmazonCloudWatch/latest/DeveloperGuide/ebs-metricscollected.html

EC2: DiskReadOps, DiskWriteOps
https://docs.aws.amazon.com/AmazonCloudWatch/latest/DeveloperGuide/ec2-metricscollected.html

### CODE REVIEW

**Comment by bergquist** [comment]
@mtanda Would you please rebase this PR on master?

**Comment by jerius** [comment]
+1 this is great! We just ran into this issue today.

**Comment by mtanda** [comment]
@bergquist I rebase on master!

And I found some bug in past PR https://github.com/grafana/grafana/pull/3066 .
The change attribute should be "onChange()".
https://github.com/grafana/grafana/pull/3445/files#diff-9036c4ac4bb1e8c05759822eaead9a70R46
Should I open another PR for fix this bug?

**Comment by bergquist** [comment]
@mtanda A new PR for that bug would be the correct way

**Comment by mtanda** [comment]
@bergquist okay, I create PR for bug fix :-)
https://github.com/grafana/grafana/pull/3531

**Comment by mtanda** [comment]
rebase again.

**Comment by mtanda** [comment]
@jerius Unfortunately, this PR is reverted by https://github.com/grafana/grafana/issues/3542.
I want more +1 vote for original issue, https://github.com/grafana/grafana/issues/3187 .
Would you add it?

**Comment by torkelo** [comment]
doing metric transformations is up to the time series database & query execution.

I want metric transformations (like sum, divide, scale, percent etc) in Grafana but has be implemented in a generic fashion so it can apply to all data sources.

**Comment by torkelo** [comment]
sorry that comment was for #3542

**Comment by mtanda** [comment]
Ah okay, this issue might not be fixed after several months...
I'll create local CloudWatch plugin fork, I use it for a while.

**Comment by torkelo** [comment]
It's possible that we have wrongly reevaluted this, reopened https://github.com/grafana/grafana/issues/3187 so we can continue discussion, maybe there is a more elegant way to solve it?

**Comment by torkelo** [comment]
@mtanda I am sorry for closing issues, you have made some very important contributions to Grafana and do not want to loose your continued support. But understand that when evaluating PRs I have to consider the long term impact of maintainability and how many really need the feature, etc. And I am open to changing my mind about this PR if the problem is big enough and encountered by many people. Since I do not use the Cloudwatch plugin for monitoring anything real it might be hard for me to judge how important this is. But I also do not want to add a hack to grafana to work around a Cloudwatch bug :)

**Comment by torkelo** [comment]
What if this was solved by having a "Sum/Period" option in the stats selection in the query editor? That fits in better conceptually. Might tricky remove this from the query that is sent to cloudwatch, but maybe worth a try? I just hate to have a hacky checkbox for this that is only applicable to a limited cases.

**Comment by mtanda** [comment]
I intend to understand your position.
Sorry, I should explain more detail what I want.
(I really like Grafana, so I really appreciate you.)

I don't have any idea to solve this elegantly.
I'll put screenshot to explain what is the problem.

**Comment by mtanda** [comment]
This is the problem what I met.

To use DynamoDB, we need to reserve computing resource before using it. (the reserved resource is called "Provisioned Throughput".)
If actual consumed throughput exceed the provisioned throughput, AWS limit the throughput.
https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/ProvisionedThroughputIntro.html

So, we need to monitor consumed / provisioned throughput.

I capture graph in our environment.
dynamodb_sum_period

Basically, the consumed throughput should not exceed provisioned throughput.
So, "Sum / Period" graph seems to be correct.
"Sum / SampleCount" graph is totally meaningless, we can't get any information how much we use provisioned throughput.

**Comment by mtanda** [comment]
I feel very sorry that I made you troubled.
I'm willing to contribute Grafana and some datasource plugin.
I want to continue a good relationship with you.

**Comment by torkelo** [comment]
no bad feelings @mtanda  :)
